luigi/task.py

```diff
@@ -146,6 +146,8 @@ class MyTask(luigi.Task):
     #: Only works when using multiple workers.
     worker_timeout = None
 
+    config = {}
```

Comment:
Wait a second ..., this is not the first time we do task-level config. Perhaps you could just follow the way previous contributors have done?

https://github.com/spotify/luigi/blob/e02ba97c8d29e61cdd7d4c7a8845dcd05ed9a35b/luigi/task.py#L133-L156

I believe it as you should do like it's done for `owner_email`, that is (1.) use `@property` and (2.) add docs under it.

This method seems much better than the dict approach. For instance you can hyperlink to config options [like this](http://luigi.readthedocs.io/en/stable/api/luigi.task.html#luigi.task.Task.owner_email)